src/main/java/net/dv8tion/jda/internal/components/selections/StringSelectMenuImpl.java

```diff
@@ -73,13 +77,20 @@ public List<SelectOption> getOptions()
         return Collections.unmodifiableList(options);
     }
 
+    @Override
+    public boolean isRequired()
+    {
+        return required;
+    }
+
     @Nonnull
     @Override
     public DataObject toData()
     {
         return super.toData()
                 .put("type", Type.STRING_SELECT.getKey())
-                .put("options", DataArray.fromCollection(options));
+                .put("options", DataArray.fromCollection(options))
+                .put("required", required);
```

Comment:
Let's only serialize it when it's explicitly set, so it works on modal and message properly.